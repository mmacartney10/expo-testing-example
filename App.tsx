import React, { useState } from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, View, TouchableOpacity, TextInput } from 'react-native';

export const INPUT_TEXT_VALID = 'valid';
export const INPUT_TEXT_INVALID = 'invalid';

export default function App() {

  const [ isLoading, setIsLoading ] = useState(false);
  const [ inputText, setInputText ] = useState("");

  const handleOnPress = () => {
    const isValid = inputText === INPUT_TEXT_VALID;
    setIsLoading(!isValid);
  }

  return (
    <View style={styles.container}>
      <StatusBar style="auto" />

      <TextInput
        onChangeText={setInputText}
        value={inputText}
        testID="input"
      />

      <TouchableOpacity onPress={handleOnPress} disabled={isLoading} testID="button">
        Click me
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
