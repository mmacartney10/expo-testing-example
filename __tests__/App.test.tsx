import React from 'react';
import {
  render,
  fireEvent,
  waitFor
} from "@testing-library/react-native";

import App, { INPUT_TEXT_INVALID, INPUT_TEXT_VALID } from '../App';

describe('When App is loaded', () => {

  let component: ReturnType<typeof render>;

  beforeEach(() => {
    component = render(<App />);
  });

  it('Should render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should not disable the button if the input has valid text', async () => {

    const input = component.getByTestId('input');
    const button = component.getByTestId('button');

    await waitFor(() => {
      fireEvent.changeText(input, INPUT_TEXT_VALID);
    });

    fireEvent.press(button);

    await waitFor(() => {
      expect(input.props.value).toBe(INPUT_TEXT_VALID);
      expect(button).toBeEnabled();
    });
  });

  it('Should disable the button if the input has invalid text', async () => {

    const input = component.getByTestId('input');
    const button = component.getByTestId('button');

    await waitFor(() => {
      fireEvent.changeText(input, INPUT_TEXT_INVALID);
    });

    fireEvent.press(button);

    await waitFor(() => {
      expect(input.props.value).toBe(INPUT_TEXT_INVALID);
      expect(button).toBeDisabled();
    });
  });
});
