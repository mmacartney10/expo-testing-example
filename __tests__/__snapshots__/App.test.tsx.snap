// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When App is loaded Should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    testID="input"
    value=""
  />
  <View
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="button"
  >
    Click me
  </View>
</View>
`;
